src/module_5/module5.md

```diff
@@ -0,0 +1,5880 @@
+    if i % period_frequency == 0:
+        print(f"Training for execution date: {execution_date}")
+        (
+            all_results,
+            all_predicted_tickers_list,
+            all_models,
+            X_train,
+            X_test,
+            all_feature_importance,
+        ) = run_model_for_execution_date(
+            execution_date,
+            all_results,
+            all_predicted_tickers_list,
+            all_models,
+            all_feature_importance,
+            params,
+        )
+```
+
+    Training for execution date: 2005-06-30T00:00:00.000000000
+    Training for execution date: 2006-06-30T00:00:00.000000000
+    Training for execution date: 2007-06-30T00:00:00.000000000
+    Training for execution date: 2008-06-30T00:00:00.000000000
+    Training for execution date: 2009-06-30T00:00:00.000000000
+    Training for execution date: 2010-06-30T00:00:00.000000000
+    Training for execution date: 2011-06-30T00:00:00.000000000
+    Training for execution date: 2012-06-30T00:00:00.000000000
+    Training for execution date: 2013-06-30T00:00:00.000000000
+    Training for execution date: 2014-06-30T00:00:00.000000000
+    Training for execution date: 2015-06-30T00:00:00.000000000
+    Training for execution date: 2016-06-30T00:00:00.000000000
+    Training for execution date: 2017-06-30T00:00:00.000000000
+    Training for execution date: 2018-06-30T00:00:00.000000000
+    Training for execution date: 2019-06-30T00:00:00.000000000
+    Training for execution date: 2020-06-30T00:00:00.000000000
+
+
+
+```python
+all_feature_importance = all_feature_importance.sort_values(
+    ["execution_date", "importance"], ascending=False
+)
+all_feature_importance_mean = (
+    all_feature_importance.groupby("feature")["importance"].mean().reset_index()
+)
+all_feature_importance_mean = all_feature_importance_mean.sort_values(
+    "importance", ascending=False
+)
+all_feature_importance_mean_importants = all_feature_importance_mean.head(10)
+```
+
+
+```python
+# Convert 'feature' to a categorical type with ordered levels
+all_feature_importance_mean_importants["feature"] = pd.Categorical(
+    all_feature_importance_mean_importants["feature"],
+    categories=all_feature_importance_mean_importants["feature"],
+    ordered=True,
+)
+
+plot = (
+    ggplot(
+        all_feature_importance_mean_importants,
+        aes(x="feature", y="importance", fill="importance"),
+    )
+    + geom_col()
+    + scale_fill_gradient(low="lightblue", high="darkblue")
+    + coord_flip()
+)  # This will make the plot horizontal
+
+print(plot)
+```
+
+
+    
+![png](module5_files/module5_72_0.png)
+    
+
+
+    
+
+
+Anormal importance given to close_0 (actual price). This means that my actual price will determine very strongly if I'll beat the index. We could argue that this might be somehow explained by momentum, but we saw in previous lessons that reverse stock splitting are causing an important data leakages for these cases.
```

Comment:
The momentum would make sense if we were comparing the price against the price of the same equity in another moment. Here is the absolute price which is somehow arbitrary